api/admin/controller/library_settings.py

```diff
@@ -235,3 +250,60 @@ def image_setting(self, setting):
             image.save(buffer, format="PNG")
             b64 = base64.b64encode(buffer.getvalue())
             return "data:image/png;base64,%s" % b64
+
+    def validate_geographic_areas(self, values):
```

Comment:
We talked about this in person; I'm recording the results of our conversation here.

This code looks correct, but there are a number of improvements you could make. This is a standalone method that will work even when there is no active HTTP request, and that gives you a lot of useful options that will make it more testable and more maintanable in the long run.

We decided not to do this immediately because there are a lot of validation methods in the controller module and it would be better to move them all out as part of the same project. I filed https://jira.nypl.org/browse/SIMPLY-1800 to cover this.

In the medium term we're going to change this so that if the validation code doesn't recognize a place, it calls out to the library registry and sees if _it_ recognizes the place. The local validation will then just be a quick way to avoid having to call out to the registry in most cases.